### PR TITLE
Doc V Capture 

### PR DESCRIPTION
**Pull request by JubrilO**
## Summary

- Remove QuadrilateralView
- Detect document aspect ratio and crop image to that aspect ratio
- Update document capture view to match UI design

### CODE REVIEW

File: Sources/SmileID/Classes/RectangleDetector/TextDetector.swift
Comment (vanshg) [inline Sources/SmileID/Classes/RectangleDetector/TextDetector.swift:1]: This should be located elsewhere; not under the `RectangleDetector` directory

File: Example/Podfile.lock [hunk 1]
```diff
@@ -1,6 +1,6 @@
 PODS:
   - netfox (1.21.0)
-  - SmileID (10.0.0-beta02):
+  - SmileID (10.0.0-beta03):
```
Comment (vanshg) [inline Example/Podfile.lock:3]: Shouldn't this be beta07?

File: Sources/SmileID/Classes/RectangleDetector/TextDetector.swift [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import Foundation
+import Vision
+import CoreImage
+
+
+class TextDetector {
+    let sequenceHandler = VNSequenceRequestHandler()
+    weak var model: DocumentCaptureViewModel?
```
Comment (vanshg) [inline Sources/SmileID/Classes/RectangleDetector/TextDetector.swift:8]: We should not couple a TextDetector to a specific ViewModel, there should be a layer of abstraction here. Something like a TextDetectorDelegate, with methods for onTextDetected and onNoTextDetected

File: Sources/SmileID/Classes/DocumentVerification/View/DocumentCaptureView.swift [hunk 3]
```diff
@@ -42,9 +42,11 @@ struct DocumentCaptureView: View {
...
                             .frame(width: 235, alignment: .center)
                     }
-                    CaptureButton {
-                        viewModel.captureImage()
-                    }.padding(.bottom, 60)
+                    if viewModel.borderColor == SmileID.theme.success.uiColor() {
+                        CaptureButton {
+                            viewModel.captureImage()
+                        }.padding(.bottom, 60)
+                    }
```
Comment (vanshg) [inline Sources/SmileID/Classes/DocumentVerification/View/DocumentCaptureView.swift:49]: The capture button visibility should be its own property -- the eventual scenario will be that the capture button is not visible for the first 10 seconds, even though the border is green. And also, we wouldn't want to show the button when a capture is in progress but we still want to show a green border

File: Sources/SmileID/Classes/DocumentVerification/Model/DocumentCaptureModel.swift [hunk 4]
```diff
@@ -333,12 +341,49 @@ class DocumentCaptureViewModel: ObservableObject, JobSubmittable, ConfirmationDi
             guard let self else {
```
Comment (vanshg) [inline Sources/SmileID/Classes/DocumentVerification/Model/DocumentCaptureModel.swift:341]: Can `self` ever be `nil`?

File: Sources/SmileID/Classes/DocumentVerification/Model/DocumentCaptureModel.swift [hunk 5]
```diff
@@ -143,30 +149,35 @@ class DocumentCaptureViewModel: ObservableObject, JobSubmittable, ConfirmationDi
...
+                let orientedImage = ciImage.oriented(forExifOrientation: Int32(cgOrientation.rawValue))
+                let scaledQuad = cropQuad.scale(CGSize(width: width, height: height), capturedImage.size)
+                var cartesianScaledQuad = scaledQuad.toCartesian(withHeight: capturedImage.size.height)
+        cartesianScaledQuad.reorganize()
+                let filteredImage = orientedImage.applyingFilter("CIPerspectiveCorrection", parameters: [
+                    "inputTopLeft": CIVector(cgPoint: cartesianScaledQuad.bottomLeft),
+                    "inputTopRight": CIVector(cgPoint: cartesianScaledQuad.bottomRight),
+                    "inputBottomLeft": CIVector(cgPoint: cartesianScaledQuad.topLeft),
+                    "inputBottomRight": CIVector(cgPoint: cartesianScaledQuad.topRight)
+                ])
```
Comment (vanshg) [inline Sources/SmileID/Classes/DocumentVerification/Model/DocumentCaptureModel.swift:172]: nit: remote extra indent

File: Sources/SmileID/Classes/DocumentVerification/Model/DocumentCaptureModel.swift (hunk 4 above)
Comment (JubrilO) [reply Sources/SmileID/Classes/DocumentVerification/Model/DocumentCaptureModel.swift:341]: Unlikely, but possible if the viewmodel is deallocated before the DispatchQueue executes this block.

File: Example/Podfile.lock (hunk 1 above)
Comment (JubrilO) [reply Example/Podfile.lock:3]: The feature branch is out of sync with main. Need to update that branch. Will do that in another PR.

**Review by vanshg — APPROVED** [review]

**Pull request merged**